src/main/java/pl/pop/interview/master/account/AccountFacade.java

```diff
@@ -1,13 +1,39 @@
 package pl.pop.interview.master.account;
 
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import pl.pop.interview.master.practitioner.Practitioner;
+import pl.pop.interview.master.practitioner.PractitionerFacade;
+
 import java.util.List;
-/**
- * The {@code AccountFacade} interface defines high-level operations for managing user accounts.
- * Implementing classes are responsible for providing the actual logic to create new accounts
- * and retrieve information about existing accounts.
- */
-public interface AccountFacade {
-    AccountDTO createNewAccount(AccountDTO accountDTO);
-
-    List<AccountDTO> getAllAccounts();
+
+@Service
+@RequiredArgsConstructor
+public class AccountFacade {
+
+    private final AccountRepository accountRepository;
+    private final PasswordEncoder passwordEncoder;
+    private final PractitionerFacade practitionerFacade;
+
+    public AccountDTO createNewAccount(AccountDTO accountDTO) {
+        if (accountRepository.existsById(accountDTO.getEmail())) {
+            throw new AccountServiceException("An account with this email address already exists");
```

Comment:
Proszę, nie róbcie nazw exceptionów jako nazwy gdszie one występują. 

to mi wygląda na "AccountException" albo "EmailInUseException" albo nawet czyste "RuntimeException"

Reply:
RuntimeException bym nie rzucał żeby nie maskowało przyczyny. AccountException też pokazuje tylko gdzie występują. Więc może ta opcja z EmailInUse?

Reply:
Runtime też będzie okej, o ile dasz ten sam message :) 
Ale EmailInUse też może być. 

Zostawiam do opinii kodującego